assets/js/components/setup/compatibility-checks.js

```diff
@@ -65,6 +66,31 @@ const checks = [
 			throw ERROR_TOKEN_MISMATCH;
 		}
 	},
+	// Check that server can connect to Google's APIs via the core/site/data/health-checks endpoint.
+	async () => {
+		try {
+			const response = await API.get( 'core', 'site', 'health-checks', undefined, {
+				useCache: false,
+			} );
+
+			if ( ! response?.checks?.googleAPI?.pass ) {
+				throw ERROR_GOOGLE_API_CONNECTION_FAIL;
+			}
+		} catch ( error ) {
+			throw ERROR_GOOGLE_API_CONNECTION_FAIL;
+		}
+	},
+	// Check that client can connect to AMP Project.
+	async () => {
+		try {
+			const response = await fetch( AMP_PROJECT_TEST_URL );
+			if ( ! response.ok ) {
+				throw ERROR_AMP_CDN_RESTRICTED;
+			}
+		} catch ( error ) {
+			throw ERROR_AMP_CDN_RESTRICTED;
+		}
```

Comment:
In this case it makes sense for a `fetch` failure to throw this error as a blocked connection at the network level will raise an error.

Please reorganize this though so that we don't have the nested throws. Instead (see above) we can have `await fetch().catch(.. throw)` and then also throw after that if `! response.ok` like you have now.